samples/extensions/conditional_rendering/conditional_rendering.cpp

```diff
@@ -0,0 +1,453 @@
+	pipeline_create_info.stageCount = static_cast<uint32_t>(shader_stages.size());
+	pipeline_create_info.pStages    = shader_stages.data();
+
+	// Vertex bindings an attributes for model rendering
+	// Binding description
+	// @todo: comment on separate buffers (gltf)
+	std::vector<VkVertexInputBindingDescription> vertex_input_bindings = {
+	    vkb::initializers::vertex_input_binding_description(0, sizeof(glm::vec3), VK_VERTEX_INPUT_RATE_VERTEX),
+	    vkb::initializers::vertex_input_binding_description(1, sizeof(glm::vec3), VK_VERTEX_INPUT_RATE_VERTEX),
+	};
+
+	// Attribute descriptions
+	std::vector<VkVertexInputAttributeDescription> vertex_input_attributes = {
+	    vkb::initializers::vertex_input_attribute_description(0, 0, VK_FORMAT_R32G32B32_SFLOAT, 0),        // Position
+	    vkb::initializers::vertex_input_attribute_description(1, 1, VK_FORMAT_R32G32B32_SFLOAT, 0),        // Normal
+	};
+
+	VkPipelineVertexInputStateCreateInfo vertex_input_state = vkb::initializers::pipeline_vertex_input_state_create_info();
+	vertex_input_state.vertexBindingDescriptionCount        = static_cast<uint32_t>(vertex_input_bindings.size());
+	vertex_input_state.pVertexBindingDescriptions           = vertex_input_bindings.data();
+	vertex_input_state.vertexAttributeDescriptionCount      = static_cast<uint32_t>(vertex_input_attributes.size());
+	vertex_input_state.pVertexAttributeDescriptions         = vertex_input_attributes.data();
+
+	pipeline_create_info.pVertexInputState = &vertex_input_state;
+
+	blend_attachment_state.blendEnable = VK_FALSE;
+	pipeline_create_info.layout        = pipeline_layout;
+	color_blend_state.attachmentCount  = 1;
+	color_blend_state.pAttachments     = blend_attachment_states.data();
+
+	shader_stages[0] = load_shader("conditional_rendering/model.vert", VK_SHADER_STAGE_VERTEX_BIT);
+	shader_stages[1] = load_shader("conditional_rendering/model.frag", VK_SHADER_STAGE_FRAGMENT_BIT);
+
+	// Flip cull mode
+	rasterization_state.cullMode = VK_CULL_MODE_NONE;
+	VK_CHECK(vkCreateGraphicsPipelines(get_device().get_handle(), pipeline_cache, 1, &pipeline_create_info, nullptr, &pipeline));
+}
+
+// Prepare and initialize uniform buffer containing shader uniforms
+void ConditionalRendering::prepare_uniform_buffers()
+{
+	// Matrices vertex shader uniform buffer
+	uniform_buffer = std::make_unique<vkb::core::Buffer>(get_device(),
+	                                                     sizeof(uniform_data),
+	                                                     VK_BUFFER_USAGE_UNIFORM_BUFFER_BIT,
+	                                                     VMA_MEMORY_USAGE_CPU_TO_GPU);
+
+	update_uniform_buffers();
+}
+
+void ConditionalRendering::update_uniform_buffers()
+{
+	uniform_data.projection = camera.matrices.perspective;
+	uniform_data.view       = camera.matrices.view * glm::mat4(1.0f);
+	// @todo: comment
+	uniform_data.view       = glm::scale(uniform_data.view, glm::vec3(0.1f, -0.1f, 0.1f));
```

Comment:
Why do you multiply camera.matrices.view with the identity, and scale it in a separate step?